R/dependecies.R

```diff
@@ -0,0 +1,437 @@
+#' Get dependencies for all elements in lock
+get_ordered_dependencies <- function(lock) {
+   cat(crayon::blue("Retrieving dependency info..."))
+   names(lock) <- vapply(lock, `[[`, character(1), "name")
+   get_dependencies_for_list(lock, lock, list())
+}
+
+#' Recursive function to take a list and lock and extract dependencies, sorting
+#' along the way using the combine_dependencies function.
+#' @param master_list list.  Packages to go through and parse dependencies from
+#' @param lock list.  Original list of locked packages
+#' @param previously_parsed_deps.  List of packages and their dependencies that
+#'   we have already parsed out of their respective description files
+get_dependencies_for_list <- function(master_list, lock, previously_parsed_deps) {
```

Comment:
Let me summarize to make sure I understand correctly. If the tree of dependencies runs to depth 5, this will recursively call itself 5 times until all the dependencies have been exhausted into the `master_list`, at which point it will have no further updates and the `identical` clause below terminates the entire chain?